homeassistant/components/packages.py

```diff
@@ -0,0 +1,76 @@
+"""
+Support for Packages.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/packages/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'packages'
+CONF_PACKAGES = DOMAIN
+
+_COMPONENT_SCHEMA = vol.Schema({cv.slug: vol.Any(dict, list)})
+
+_CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({cv.slug: _COMPONENT_SCHEMA})
+}, extra=vol.ALLOW_EXTRA)
+
+
+def _log_error(package, component, message, *message_args):
+    """Log an error while merging."""
+    _LOGGER.error("Package %s setup failed. Component %s %s",
+                  package, component, message.format(*message_args))
+
+
+def merge_packages_config(config):
+    """Merge packages into the root level config. Mutate config."""
+    if CONF_PACKAGES not in config:
+        return config
+
+    _CONFIG_SCHEMA(config)
+
+    blocked_comps = ['homeassistant']
+    for pack_name, pack_conf in config[CONF_PACKAGES].items():
```

Comment:
Let's have this live under `homeassistant` as it's not a component but related to running Home Assistant.